### PR TITLE
samba: 4.7.10 -> 4.7.12

### PR DESCRIPTION
**Pull request by andrew-d**
###### Motivation for this change
Security update; see #55392

**EDIT:**

I don't know if this is going to be fixed with the most recent force-push, but I'm seeing a bunch of the following errors:
```
anonymous function at [...]pkgs/applications/networking/remote/x2goclient/default.nix:1:1 called without required argument 'nxproxy', at [...]/lib/customisation.nix:69:12
```

This doesn't seem at all related to the Samba changes in this PR, so I have no idea what's going on :man_shrugging: 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc? @ckauhaus

### CODE REVIEW

**Comment by andrew-d** [comment]
I don't know if this is going to be fixed with the most recent force-push, but I'm seeing a bunch of the following errors:
```
anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-0-gleber.ewr1.nix.ci/pkgs/applications/networking/remote/x2goclient/default.nix:1:1 called without required argument 'nxproxy', at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-0-gleber.ewr1.nix.ci/lib/customisation.nix:69:12
```

This doesn't seem at all related to the Samba changes in this PR, so I have no idea what's going on :man_shrugging:

**Comment by srhb** [comment]
@GrahamcOfBorg eval

**Review by bachp — APPROVED** [review]
Sucessfully tested with Windows 10 for several days

**Comment by xeji** [comment]
@grahamcofborg build samba sambaFull

**Comment by xeji** [comment]
Darwin build should be fixed before we merge this.

**Comment by andrew-d** [comment]
@xeji - Any idea what the error is?  Both builds appear to fail just after the `patching script interpreter paths` message, but I don't see any error messages that say why that is...

**Comment by xeji** [comment]
It also failed on x64_64-linux this time. 
Looking at the full logs, there are several compiler errors both for Darwin and x64_64-linux. 
Not sure about the cause.

**Comment by vcunat** [comment]
For the `sambaFull` problem see: https://github.com/NixOS/nixpkgs/pull/55893#issuecomment-466740831

**Comment by vcunat** [comment]
`samba` itself builds fine for me.

**Comment by andrew-d** [comment]
@vcunat - I just force-pushed a change that fixes the `sambaFull` build locally

**Comment by vcunat** [comment]
So the plan is to have `sambaFull` contain the "insecure" combination and hope for admin not to utilize it or something.  Well, `sambaFull` isn't used by default anywhere, so I expect that will be OK, though I can't say I understand stuff around running a samba service.

**Comment by vcunat** [comment]
@GrahamcOfBorg build samba sambaFull

**Review by vcunat — APPROVED** [review]
Same as for 18.09, tested on two platforms, builds of two packages and the nixos test.

**Comment by vcunat** [comment]
Notes:
- we have no `meta.maintainers`, even though the package/service seems relatively important (I imagine);
- release notes for 4.7.11 suggested that the 4.7 branch is about its EOL, so it might be a good time to upgrade, possibly before 19.03 fork-off, though security patches will probably still be available, so I expect it's not too urgent.

**Comment by andrew-d** [comment]
@vcunat - Thank you for the help here!

**Pull request merged**
